File: java/sdk/sdk-pay/src/main/java/com/novax/sdk/pay/h5/H5AddressRequest.java
package com.novax.sdk.pay.h5;

import com.novax.sdk.core.request.AbstractApiRequest;
import com.novax.sdk.core.request.HttpMethod;
import com.novax.sdk.core.request.TypeRef;
import com.novax.sdk.pay.model.PayOrderAddressResp;
import lombok.Builder;
import lombok.Getter;

import java.util.LinkedHashMap;
import java.util.Map;

/**
 * GET /pay/public/h5/address — get the H5 payment receipt address.
 *
 * <p>Server expects {@code language} header (already injected by
 * {@code HeaderInterceptor} if configured on the client).
 */
@Getter
@Builder
public class H5AddressRequest extends AbstractApiRequest<PayOrderAddressResp> {

    private final String protocol;
    private final String currency;
    private final String smartContractAddress;
    private final String companyUserId;
    private final String token;

    @Override public HttpMethod method() { return HttpMethod.GET; }
    @Override public String path() { return "/pay/public/h5/address"; }

    @Override public Map<String, ?> queryParams() {
        Map<String, Object> q = new LinkedHashMap<>();
        if (protocol != null) q.put("protocol", protocol);
        if (currency != null) q.put("currency", currency);
        if (smartContractAddress != null) q.put("smartContractAddress", smartContractAddress);
        if (companyUserId != null) q.put("companyUserId", companyUserId);
        if (token != null) q.put("token", token);
        return q;
    }

    @Override public TypeRef<PayOrderAddressResp> responseType() {
        return TypeRef.of(PayOrderAddressResp.class);
    }
}

File: python/novax_sdk/pay/h5/h5_address_request.py
from dataclasses import dataclass
from typing import Any

from novax_sdk.pay.model import PayOrderAddressResp
from novax_sdk.request.api_request import ApiRequest, HttpMethod


@dataclass
class H5AddressRequest(ApiRequest):
    """GET /pay/public/h5/address"""

    protocol: str | None = None
    currency: str | None = None
    smart_contract_address: str | None = None
    company_user_id: str | None = None
    token: str | None = None

    @property
    def method(self) -> HttpMethod:
        return HttpMethod.GET

    @property
    def path(self) -> str:
        return "/pay/public/h5/address"

    def query_params(self) -> dict[str, Any]:
        return {k: v for k, v in {
            "protocol": self.protocol,
            "currency": self.currency,
            "smartContractAddress": self.smart_contract_address,
            "companyUserId": self.company_user_id,
            "token": self.token,
        }.items() if v is not None}

    @property
    def response_type(self) -> type:
        return PayOrderAddressResp
